rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -230,43 +265,47 @@ rosbag2_storage::BagMetadata SqliteStorage::get_metadata()
 {
   rosbag2_storage::BagMetadata metadata;
   metadata.storage_identifier = "sqlite3";
-  metadata.relative_file_paths = {database_name_};
+  metadata.relative_file_paths = {database_files_};
 
   metadata.message_count = 0;
   metadata.topics_with_message_count = {};
 
-  auto statement = database_->prepare_statement(
-    "SELECT name, type, serialization_format, COUNT(messages.id), MIN(messages.timestamp), "
-    "MAX(messages.timestamp) "
-    "FROM messages JOIN topics on topics.id = messages.topic_id "
-    "GROUP BY topics.name;");
-  auto query_results = statement->execute_query<
-    std::string, std::string, std::string, int, rcutils_time_point_value_t,
-    rcutils_time_point_value_t>();
-
   rcutils_time_point_value_t min_time = INT64_MAX;
   rcutils_time_point_value_t max_time = 0;
-  for (auto result : query_results) {
-    metadata.topics_with_message_count.push_back(
-      {
-        {std::get<0>(result), std::get<1>(result), std::get<2>(result)},
-        static_cast<size_t>(std::get<3>(result))
-      });
-
-    metadata.message_count += std::get<3>(result);
-    min_time = std::get<4>(result) < min_time ? std::get<4>(result) : min_time;
-    max_time = std::get<5>(result) > max_time ? std::get<5>(result) : max_time;
-  }
+  for (const std::string & database_name : database_files_) {
+    std::unique_ptr<SqliteWrapper> database = std::make_unique<SqliteWrapper>(
+      rosbag2_storage::FilesystemHelper::concat({uri_, database_name}), io_flag_);
+    auto statement = database->prepare_statement(
+      "SELECT name, type, serialization_format, COUNT(messages.id), MIN(messages.timestamp), "
+      "MAX(messages.timestamp) "
+      "FROM messages JOIN topics on topics.id = messages.topic_id "
+      "GROUP BY topics.name;");
+    auto query_results = statement->execute_query<
+      std::string, std::string, std::string, int, rcutils_time_point_value_t,
+      rcutils_time_point_value_t>();
+
+    for (auto result : query_results) {
+      metadata.topics_with_message_count.push_back(
+        {
+          {std::get<0>(result), std::get<1>(result), std::get<2>(result)},
+          static_cast<size_t>(std::get<3>(result))
+        });
+
+      metadata.message_count += std::get<3>(result);
+      min_time = std::get<4>(result) < min_time ? std::get<4>(result) : min_time;
+      max_time = std::get<5>(result) > max_time ? std::get<5>(result) : max_time;
+    }
 
-  if (metadata.message_count == 0) {
-    min_time = 0;
-    max_time = 0;
+    if (metadata.message_count == 0) {
+      min_time = 0;
+      max_time = 0;
+    }
+    metadata.bag_size += rosbag2_storage::FilesystemHelper::calculate_directory_size(database_name);
```

Comment:
This is **directory** size, not file size - so do we really need to add the size of all databases?